### PR TITLE
✨⚡️🎨 Refactor GeoController to LocationController, add support for live location

### LINKED ISSUE #1113: Active Statuses page is slow and sluggish.
Die "Unterwegs"-Seite ist ganz schön langsam und unperformant, wenn viele Leute (z.B. `>50`) unterwegs sind. Mir sind ein paar Dinge aufgefallen:

1. Der Seitenaufbau ist ziemlich langsam (bei mir gerade `>13s`)
2. Die Seite alleine (ohne Assets) ist knapp 7MB groß, davon 6.5MB durch Pfad-Daten.
3. Die Leaflet-Karte und das Path Rendering verbraten unendlich viele CPU-Cycles. Wenn man die Karte aus dem HTML rauslöscht, sind meine Rechner damit zufrieden.

Dafür sieht die Karte so halt auch ziemlich nice aus.

Aktuelle Livekarte mit sehr sehr vielen Linien

Der Code für die Livekarte ist wahrscheinlich seit 3 Jahren nicht mehr angefasst worden, als wir noch sehr viel weniger User hatten. Ich hab noch keine sinnvolle Idee, wie wir den Load reduzieren können, bin aber für alles offen.

### LINKED ISSUE #1290: Active statuses page keeps crashing due to JS issue
When I open the page with the active statuses, I get 20+ JS error messages after a few seconds which causes the page to crash.

```
TypeError: undefined has no properties
    a ActiveJourneys.js:75
    a ActiveJourneys.js:39
    value ActiveJourneys.js:180
    setInterval handler*value ActiveJourneys.js:179
    <anonymous> active:636
```

Steps to reproduce:
- Open the "Active statuses" page in e. g. Firefox
- Wait 10-15 seconds

### LINKED ISSUE #1432: “Unterwegs” page very laggy, constantly throwing errors and high memory consumption
Page /statuses/active throws a lot of errors constantly (tens to hundreds per second), uses a lot of memory and is very laggy in general.

Errors include:
```
TypeError: undefined has no properties
    a ActiveJourneys.js:75
    a ActiveJourneys.js:39
    value ActiveJourneys.js:180
    setInterval handler*value ActiveJourneys.js:179
    <anonymous> active:793
    EventListener.handleEvent* active:188
```

```
TypeError: e.stopovers[a.length] is undefined
    a ActiveJourneys.js:75
    a ActiveJourneys.js:39
    value ActiveJourneys.js:178
    <anonymous> active:793
```

```
TypeError: e.stopovers[y] is undefined
    a ActiveJourneys.js:131
    a ActiveJourneys.js:39
    value ActiveJourneys.js:180
```

The Page uses about 340 MiB of RAM (which is A LOT in comparison to other websites like github.com with 20 MiB or traeweling.com/dashboard with 15 MiB), almost 200 MiB of that is in domNode.path.

  
Screenshot_20230305_171218

Using that page is not a smooth user experience at all, it takes long to load, is laggy and sometimes crashes the browser.

It probably has to do with the Map.

### PR DESCRIPTION
**Pull request by HerrLevin**
Closes #1432, closes #1290, closes #1113

### CODE REVIEW

File: app/Dto/GeoJson/Feature.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    public function toArray(): array {
+        $response = [
+            'type'     => 'Feature',
+            'geometry' => [
+                'type'        => $this->type,
+                'coordinates' => $this->coordinates
+            ],
+
+
```
Comment (MrKrisKrisu) [inline app/Dto/GeoJson/Feature.php]: please remove empty lines

File: app/Http/Controllers/API/v1/StatusController.php [hunk 2]
```diff
@@ -176,7 +180,78 @@ public static function getFutureCheckins(): AnonymousResourceCollection {
...
+     *     ),
+     *     @OA\Response(response=403, description="User not authorized to access this status"),
+     *       security={
+     *           {"passport": {"read-statuses"}}, {"token": {}}
+     *
+     *       }
+     *     )
+     * )
+     */
+    public function livePositions() {
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/API/v1/StatusController.php]: Return type?

File: app/Http/Controllers/API/v1/StatusController.php [hunk 3]
```diff
@@ -176,7 +180,78 @@ public static function getFutureCheckins(): AnonymousResourceCollection {
...
+     *     ),
+     *     @OA\Response(response=403, description="User not authorized to access this status"),
+     *       security={
+     *           {"passport": {"read-statuses"}}, {"token": {}}
+     *
+     *       }
+     *     )
+     * )
+     */
+    public function getLivePositionForStatus($ids) {
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/API/v1/StatusController.php]: Return type?

File: app/Http/Controllers/Backend/Support/LocationController.php [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+    public static function forStatus(Status $status): LocationController {
+        return new self(
+            $status->trainCheckin->HafasTrip,
+            $status->trainCheckin->origin_stopover,
+            $status->trainCheckin->destination_stopover,
+            $status->id
+        );
+    }
+
+    private function filterStopOversFromStatus(): ?array {
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/Backend/Support/LocationController.php]: we call it "Stopover" everywhere. So I would write the `o` in lower case.

File: app/Http/Controllers/Backend/Support/LocationController.php [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+     */
+    public function calculateLivePosition(): ?LivePointDto {
+        $hafasTrip    = $this->hafasTrip;
+        $newStopovers = $this->filterStopOversFromStatus();
+
+        if (!$newStopovers) {
+            return null;
+        }
+        if (count($newStopovers) === 1) {
+
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/Backend/Support/LocationController.php]: empty line

File: app/Http/Controllers/Backend/Support/LocationController.php [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+                )),
+                null,
+                $newStopovers[0]->arrival->timestamp,
+                $newStopovers[0]->departure->timestamp,
+                $hafasTrip->linename,
+                $this->statusId
+            );
+        }
+
+        $now               = Carbon::now()->timestamp;
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/Backend/Support/LocationController.php:91]: funfact: you can just call `now()` without the `Carbon::`.

File: app/Http/Controllers/Backend/Support/LocationController.php [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+        $recentPoint = null;
+        $distance    = 0;
+        foreach ($polyline->features as $key => $point) {
+            $point = Coordinate::fromGeoJson($point);
+            if ($recentPoint !== null && $point !== null) {
+                $lineSegment = new LineSegment($recentPoint, $point);
+
+                $distance += $lineSegment->calculateDistance();
+                if ($distance >= $meters) {
+
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/Backend/Support/LocationController.php]: empty line

File: app/Http/Controllers/Backend/Support/LocationController.php [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+                    break;
+                }
+            }
+
+            $recentPoint = $point ?? $recentPoint;
+        }
+
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/Backend/Support/LocationController.php:113]: 2 empty lines

File: app/Http/Controllers/Backend/Support/LocationController.php [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+
+        $pointS = $lineSegment->interpolatePoint($meters / $distance);
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/Backend/Support/LocationController.php]: What is the S for?

File: app/Http/Controllers/FrontendStatusController.php [hunk 10]
```diff
@@ -53,12 +53,11 @@ public function getGlobalDashboard(): Renderable {
...
 
     public function getActiveStatuses(): Renderable {
-        $activeStatusesResponse = StatusBackend::getActiveStatuses();
         $activeEvents           = EventBackend::activeEvents();
         return view('activejourneys', [
             'currentUser' => Auth::user(),
-            'statuses'    => $activeStatusesResponse['statuses'],
-            'polylines'   => $activeStatusesResponse['polylines'],
+            'statuses'    => StatusBackend::getActiveStatuses(),
+            'polylines'   => [],
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/FrontendStatusController.php]: Can you remove the need in the frontend, so this variable can be removed completly?

**Review by MrKrisKrisu — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Backend/Support/LocationController.php (hunk 6 above)
Comment (HerrLevin) [reply app/Http/Controllers/Backend/Support/LocationController.php:91]: huh, nice. I just copied that section from the GeoController - but good to know! :D

File: app/Http/Controllers/Backend/Support/LocationController.php (hunk 9 above)
Comment (HerrLevin) [reply app/Http/Controllers/Backend/Support/LocationController.php]: Remnant from prototyping. Whoops.

**Pull request merged**